tests/e2e/specs/setup-mc/step-1-accounts.test.js

```diff
@@ -276,27 +294,361 @@ test.describe( 'Set up accounts', () => {
+
+							// Mock Merchant Center as not connected
+							setUpAccountsPage.mockMCNotConnected(),
+						] );
+
+						await page.reload();
+
+						// Mock Merchant Center create accounts
+						await setUpAccountsPage.mockMCCreateAccountWebsiteClaimed(
+							12345,
+							host
+						);
+
+						// Click "Create account" button from the page.
+						const createAccountButton =
+							setUpAccountsPage.getMCCreateAccountButtonFromPage();
+						await createAccountButton.click();
+						await page.waitForLoadState(
+							LOAD_STATE.DOM_CONTENT_LOADED
+						);
+
+						// Check the checkbox to accept ToS.
+						const modalCheckbox =
+							setUpAccountsPage.getModalCheckbox();
+						await modalCheckbox.click();
+
+						// Click "Create account" button from the modal.
+						const createAccountButtonFromModal =
+							setUpAccountsPage.getMCCreateAccountButtonFromModal();
+						await createAccountButtonFromModal.click();
+						await page.waitForLoadState(
+							LOAD_STATE.DOM_CONTENT_LOADED
+						);
+
+						const reclaimButton =
+							setUpAccountsPage.getReclaimMyURLButton();
+						await expect( reclaimButton ).toBeVisible();
+
+						const switchAccountButton =
+							setUpAccountsPage.getSwitchAccountButton();
+						await expect( switchAccountButton ).toBeVisible();
+
+						const reclaimingURLInput =
+							setUpAccountsPage.getReclaimingURLInput();
+						await expect( reclaimingURLInput ).toHaveValue(
+							baseURL
+						);
+
+						const continueButton =
+							setUpAccountsPage.getContinueButton();
+						await expect( continueButton ).toBeDisabled();
+					} );
+
+					test( 'click "Reclaim my URL" should send a claim overwrite request and see Merchant Center "Connected"', async () => {
+						await Promise.all( [
+							// Mock Merchant Center accounts claim overwrite
+							setUpAccountsPage.mockMCAccountsClaimOverwrite(
+								12345
+							),
+
+							// Mock Merchant Center as connected with ID 12345
+							setUpAccountsPage.mockMCConnected( 12345 ),
+						] );
+
+						const reclaimButton =
+							setUpAccountsPage.getReclaimMyURLButton();
+						await reclaimButton.click();
+						await page.waitForLoadState(
+							LOAD_STATE.DOM_CONTENT_LOADED
+						);
+
+						const mcConnectedLabel =
+							setUpAccountsPage.getGoogleComboConnectedLabel();
+						await expect( mcConnectedLabel ).toContainText(
+							'Connected'
+						);
+
+						const mcDescriptionRow =
+							setUpAccountsPage.getMCDescriptionRow();
+						await expect( mcDescriptionRow ).toContainText(
+							'Merchant Center ID: 12345'
+						);
+					} );
+				} );
+			} );
+		} );
+
+		test.describe( 'Merchant Center has existing accounts', () => {
+			test.beforeAll( async () => {
+				await Promise.all( [
+					// Mock Jetpack as connected.
+					setUpAccountsPage.mockJetpackConnected(),
+
+					// Mock google as connected.
+					setUpAccountsPage.mockGoogleConnected(
+						'google@example.com'
+					),
```

Comment:
These mocks seem to be redundant as the upper `test.beforeAll` block has set the same mocks and they won't invalidated when entering here.

https://github.com/woocommerce/google-listings-and-ads/blob/8fba3e4ecd1d8b12f365706545d47f66976a1770/tests/e2e/specs/setup-mc/step-1-accounts.test.js#L297-L304

Reply:
👍